File: go/gin/internal/modules/branch/controller.go
package branch

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

type BranchController struct{
	service BranchService
}

func (c *BranchController) FindAll(ctx *gin.Context){
	branchs, err := c.service.FindAll()

	if err != nil {
		ctx.JSON(err.StatusCode, err)
		return
	}

	ctx.JSON(http.StatusOK, branchs)
}

File: rust/actix/src/modules/branch/controller.rs
use actix_web::web::Json;

use crate::modules::branch::entity::BranchEntity;
use crate::common::errors::base::BaseError;
use super::service::BranchService;

pub struct BranchController;

impl BranchController {
    pub async fn find_all() -> Result<Json<Vec<BranchEntity>>, BaseError> {
        match BranchService::find_all().await {
            Ok(branches) => Ok(Json(branches)),
            Err(err) => Err(err),
        }
    }
}
